src/main/java/pl/javastart/task/Main.java

```diff
@@ -1,16 +1,50 @@
 package pl.javastart.task;
 
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
 import java.util.Scanner;
 
 public class Main {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
 
     public static void main(String[] args) {
         Main main = new Main();
         main.run(new Scanner(System.in));
     }
 
     public void run(Scanner scanner) {
-        // uzupełnij rozwiązanie. Korzystaj z przekazanego w parametrze scannera
+        String[] patterns = {"yyyy-MM-dd HH:mm:ss", "yyyy-MM-dd", "dd.MM.yyyy HH:mm:ss"};
+
+        System.out.println("Podaj datę:");
+        String inputDate = scanner.nextLine();
+        for (String form : patterns) {
+            LocalDateTime dateTime;
+            try {
+                if (form.equals("yyyy-MM-dd")) {
+                    dateTime = LocalDateTime.parse(inputDate + " 00:00:00", FORMATTER);
+                } else {
+                    dateTime = LocalDateTime.parse(inputDate, DateTimeFormatter.ofPattern(form));
+                }
```

Comment:
Zamiast sprawdzać ten warunek dla każdego elementu tablicy, może lepiej dodaj sobie informację o czasie do inputu, jeżeli go brakuje (najprościej możesz to stwierdzić na podstawie długości wprowadzonego tekstu)